### PR TITLE
Modified VG/VGC_SetVariables targets to check if NodeExecutable inclu…

### PR DESCRIPTION
**Pull request by Jmales**
…des exist before executing them later on.

### CODE REVIEW

**Review by joaompneves — APPROVED** [review]

File: Example.Avalonia/Example.Avalonia.csproj [hunk 1]
```diff
@@ -27,11 +27,11 @@
   <ItemGroup>
     <PackageReference Include="Avalonia.Desktop" Version="0.10.0-preview6" />
     <PackageReference Include="NLog" Version="4.6.2" />
-    <PackageReference Include="ViewGenerator" Version="1.0.307">
+    <PackageReference Include="ViewGenerator" Version="1.0.308">
```
Comment (alvesmiguel1) [inline Example.Avalonia/Example.Avalonia.csproj:30]: You will break the build if you just increment the example's version right away

Comment (Jmales) [reply Example.Avalonia/Example.Avalonia.csproj:30]: Yeah, I incremented ViewGenerator/ViewGeneratorCore on Friday. These versions are already on Azure ;)

File: ViewGeneratorCore/build/ViewGeneratorCore.targets [hunk 2]
```diff
@@ -24,16 +24,16 @@
...
           <_ViewGeneratorCoreNodeJsExe Include="$(MSBuildThisFileDirectory)..\..\Node.js.redist.$(NodeVersion)\tools\osx-x64\node"/>
-
       </ItemGroup>
-      
+      <ItemGroup>
+          <_ViewGeneratorCoreNodeJsExeAfterCheckedPaths Include="@(_ViewGeneratorCoreNodeJsExe)" Condition="Exists('%(FullPath)')"  />
+      </ItemGroup>
     <PropertyGroup>
-      <__ViewGeneratorCoreNodeJsExe>@(_ViewGeneratorCoreNodeJsExe);</__ViewGeneratorCoreNodeJsExe>
+      <__ViewGeneratorCoreNodeJsExe>@(_ViewGeneratorCoreNodeJsExeAfterCheckedPaths);</__ViewGeneratorCoreNodeJsExe>
```
Comment (alvesmiguel1) [inline ViewGeneratorCore/build/ViewGeneratorCore.targets:36]: You need to do a substring right? You can have both file existing in filesystem

File: ViewGenerator/build/ViewGenerator.targets [hunk 3]
```diff
@@ -20,19 +20,19 @@
...
-      
+      <ItemGroup>
+          <_ViewGeneratorCoreNodeJsExeAfterCheckedPaths Include="@(_ViewGeneratorCoreNodeJsExe)" Condition="Exists('%(FullPath)')"  />
+      </ItemGroup>
     <PropertyGroup>
       <ViewGeneratorPath>$([System.IO.Path]::GetFullPath(&quot;$(MSBuildThisFileDirectory)..\&quot;))</ViewGeneratorPath>
       <ViewGeneratorToolsPath>$(ViewGeneratorPath)tools</ViewGeneratorToolsPath>
       <ViewGeneratorContentFilesPath>$(ViewGeneratorPath)contentFiles\</ViewGeneratorContentFilesPath> 
-      <__ViewGeneratorNodeJsExe>@(_ViewGeneratorNodeJsExe);</__ViewGeneratorNodeJsExe>
+      <__ViewGeneratorNodeJsExe>@(_ViewGeneratorCoreNodeJsExeAfterCheckedPaths);</__ViewGeneratorNodeJsExe>
```
Comment (alvesmiguel1) [inline ViewGenerator/build/ViewGenerator.targets:35]: Same as ViewGeneratorCore comments

**Review by alvesmiguel1 — CHANGES_REQUESTED** [review]
_

File: Example.Avalonia/Example.Avalonia.csproj (hunk 1 above)
Comment (alvesmiguel1) [reply Example.Avalonia/Example.Avalonia.csproj:30]: 👍

File: ViewGeneratorCore/build/ViewGeneratorCore.targets (hunk 2 above)
Comment (Jmales) [reply ViewGeneratorCore/build/ViewGeneratorCore.targets:36]: We are still doing that in the line below with **__ViewGeneratorCoreNodeJsExe.Substring**

Comment (alvesmiguel1) [reply ViewGeneratorCore/build/ViewGeneratorCore.targets:36]: Oh... Weird pattern, we're declaring the same property twice? Does msbuild complain about that?

Comment (alvesmiguel1) [reply ViewGeneratorCore/build/ViewGeneratorCore.targets:36]: Right, I was confused with the variable naming

**Review by alvesmiguel1 — APPROVED** [review]

**Pull request merged**